index.js

```diff
@@ -138,47 +182,66 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
+let ticketNumber = 0
 
+function takeATicketNumber(line){
+  ticketNumber += 1
+  line.push(ticketNumber)
+  return `Welcome. You are ticket number ${line.length}`
+}
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+ console.log(takeATicketNumber(line))
+ // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+ console.log(takeATicketNumber(line))
+ // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+/ console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+ console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+
+function ticketNumberGeneratorFunc(){
+
+  let number = 0
+
+  return function (line){
+    number += 1
+    line.push (number)
+    return `Welcome. You are ticket number ${number}`
+  }
+
+}
+
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+ console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+ const newLine = []
+ const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+ console.log(takeATicketNumberFunc(newLine))
+ // => `Welcome. You are ticket number 1`
```

Comment:
Great job solving this problem one way and then being able to refactor your code and find a different solution using closures. Seeing multiple paths to an outcome is not easy.